contrib/roles/windows/openvswitch/tasks/install_custom_ovs.yml

```diff
@@ -21,6 +21,16 @@
     dest: "{{ovs_info.tmp_dir}}\\certificate.cer"
     timeout: 60
   retries: 3
+  when: ovs_certs_link is defined
+
+- name: OVS | Extract certificate from msi
+  win_shell: |
+    $driverFile = "{{ovs_info.tmp_dir}}\\ovs.msi"
+    $outputFile = "{{ovs_info.tmp_dir}}\\certificate.cer"
+    $exportType = [System.Security.Cryptography.X509Certificates.X509ContentType]::Cert
+    $cert = (Get-AuthenticodeSignature $driverFile).SignerCertificate
+    [System.IO.File]::WriteAllBytes($outputFile, $cert.Export($exportType))
```

Comment:
Noob question: what if we do not write the cert file? Will the software not function?
Also is this step only for script based install (as against a GUI based click-install)?

Reply:
This is just for script based installation. When you install it via GUI there will be a pop-up asking you to accept the certificate as trusted if you want to install the software.

To avoid getting the pop-up we need to add the certificate to the trusted certs before installing OVS. Removing this step will cause ``Installing OVS`` to fail.
For the release which is signed by Microsoft this step is not required at all since you will not receive any pop-ups related to certificates.